src/pages/workspaces/workspace/analysis/AppLauncher.js

```diff
@@ -169,16 +168,8 @@ const ApplicationLauncher = _.flow(
       const localSafeModeBaseDirectory = ``
 
 
-      const loadAzureStorage = withErrorReporting('Error loading Azure storage information.', async () => {
-        const { storageContainerName } = await Ajax(signal).AzureStorage.details(workspaceId)
-        setAzureStorage(storageContainerName)
-      })
-
-      if (azureContext) {
-        await loadAzureStorage()
-      }
-
-      const cloudStorageDirectory = azureContext ? `${azureStorage}/analyses` : `gs://${bucketName}/notebooks`
+      const { storageContainerName: azureStorageContainer } = !!azureContext ? await Ajax(signal).AzureStorage.details(workspaceId) : {}
+      const cloudStorageDirectory = !!azureContext ? `${azureStorageContainer}/analyses` : `gs://${bucketName}/notebooks`
 
       //TODO: fix this when relay is working https://broadworkbench.atlassian.net/browse/IA-3700
       !!googleProject ?
```

Comment:
Would this function be easier to read if it was divided into `if (googleProject) { ... }` and `if (azureContext) { ... }` blocks instead of having a condition on every line?

Reply:
Also https://broadworkbench.atlassian.net/browse/IA-3700, which has no description in it, is marked as Done. So should this code actually be deleted now?